### PR TITLE
修正 prettyUrl 不完整

### PR DESCRIPTION
**Pull request by cyfung1031**
自己写的 prettyUrl 要自己维护

我还是接受不了只是用一个小小的工具就要引入 整个 "punycode" package
"punycode" 没有 typescript 支持, 又不是 ESM

这些工具都放在 `@App/pkg/utils/`
有unit test 就行，日后也不用改什么

实际打包后有 rspack + ES2020 / ES2022，体积和语法都是最优化

### CODE REVIEW

**Comment by CodFrm** [comment]
哎呀，我只是喜欢有现成的库尽量用现成的，不自己维护，不过也确实只是需要decode功能，引入了一堆其它的

你弄了，那就用你的吧

File: src/pkg/utils/url-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    it.concurrent("should decode CJK characters in the pathname", () => {
+      // %E6%B5%8B%E8%AF%95 -> 测试
+      expect(prettyUrl("https://example.com/%E6%B5%8B%E8%AF%95")).toBe("https://example.com/测试");
+    });
+
+    it.concurrent("should decode spaces and common symbols in path", () => {
+      expect(prettyUrl("https://site.com/hello%20world")).toBe("https://site.com/hello world");
+    });
+
+    it.concurrent("should NOT decode if it.concurrent introduces reserved URL delimiters like ? or #", () => {
```
[the remark this answers is not in the document]
Comment (cyfung1031) [reply src/pkg/utils/url-utils.test.ts]: 哈哈
我用replace all 結果連英文的 it 都變了 it.concurrent

File: src/pkg/utils/url-utils.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  try {
+    const u = new URL(s, baseUrl);
+
+    // 1. Core components: Protocol, Punycode Host, and Port
+    const protocol = u.protocol ? `${u.protocol}//` : "";
+    const host = u.hostname
+      .split(".")
+      .map((p) => (p.startsWith("xn--") ? decodePunycode(p) : p))
```
[the remark this answers is not in the document]
Comment (cyfung1031) [reply src/pkg/utils/url-utils.ts]: 有道理！

File: src/pkg/utils/url-utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,256 @@
+import { describe, it, expect } from "vitest";
+import { prettyUrl } from "./url-utils"; // Update with your actual file path
```
[the remark this answers is not in the document]
Comment (cyfung1031) [reply src/pkg/utils/url-utils.test.ts]: AI寫的沒注意到

File: src/pkg/utils/url-utils.test.ts [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    });
+  });
+
+  describe.concurrent("Edge Cases and Safety", () => {
+    it.concurrent("should return empty string for null/undefined", () => {
+      expect(prettyUrl(null as any)).toBe("");
+      expect(prettyUrl(undefined)).toBe("");
+    });
+
+    it.concurrent("should return the original string if it.concurrent is not a valid URL", () => {
```
[the remark this answers is not in the document]
Comment (cyfung1031) [reply src/pkg/utils/url-utils.test.ts]: https://github.com/scriptscat/scriptcat/pull/1244/commits/927c66d15e796fb96eff2f64d7016c5fff979758

File: src/pkg/utils/url-utils.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cyfung1031) [reply src/pkg/utils/url-utils.test.ts]: https://github.com/scriptscat/scriptcat/pull/1244/commits/927c66d15e796fb96eff2f64d7016c5fff979758
